config.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2015 Brian J. Downs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openweathermap
+
+import (
+	"errors"
+	"os"
+)
+
+// Config will hold default settings to be passed into the
+// "NewCurrent, NewForecast, etc}" functions.
+type Config struct {
+	Mode     string // user choice of JSON or XML
+	Unit     string // measurement for results to be displayed.  F, C, or K
+	Lang     string // should reference a key in the LangCodes map
+	APIKey   string // API Key for connecting to the OWM
+	Username string // Username for posting data
+	Password string // Pasword for posting data
+}
+
+// CheckAPIKeyExists will see if an API key has been set.
+func (c *Config) CheckAPIKeyExists() bool {
+	return len(c.APIKey) > 1
+}
+
+// config will hold default settings to be passed into the OWM requests
+var config = Config{}
```

Comment:
It'd be nice to initialize the Config struct and work with a pointer to it since there's only 1 config at any given time.  We then have a receiver on that pointer to set values to.

```go
c := &Config{
    Some: "other data",
}
// do the work to get the API key
c.APIKey = apiKey
// or
c.AddAPIKey(apiKey) // this would do all the checking of the passed in key
// or
c.SetKey() // this would do all the checking for the key and hte key itself and then set it on the config
```

I'd like to avoid formal getters and setters.  We can keep functionality that's specific to something, in this case config, specific to that rather than putting generic getters and setters into scope.  Ie, `GetPassword()` etc.  Get password for what? For whom and to be used in what manner?  We'd have to reference the implementation to see that.  If this is just part of the initialization, it's clear what we're doing with what and why.